packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -156,16 +189,97 @@ function DefaultSelectionTracker(props: DefaultSelectionTrackProps) {
     }
   }, [])
 
+  // First: get the current bounds of the element
+  useEffect(() => {
+    if (domRef.current) {
+      let item = domRef.current.querySelector("label[data-selected=true]");
+      itemRef.current = item;
+
+      // i don't really want to have this here, and honestly, it should probably go in useLayoutEffect
+      // however, then i basically have to pull all the code currently in useEffect to useLayoutEffect which feels repetitive so I'm just keeping it here for now
+      let styleObj: { transform: string | undefined, width: string | undefined, height: string | undefined } = {
+        transform: undefined,
+        width: undefined,
+        height: undefined
+      };
+
+      let finalW = item?.offsetWidth;
+      let finalH = item?.offsetHeight;
+
+      styleObj.width = finalW;
+      styleObj.height = finalH;
+      setStyle(styleObj);
+
+    }
+  }, [state?.selectedValue])
+
+  // Last: get the final bounds of the element
+  useLayoutEffect(() => {
+    if (domRef.current) {
+      let slide = document.querySelector('#animate');
+      let item = domRef.current.querySelector("label[data-selected=true]")
+      let cachedItem = itemRef?.current?.getBoundingClientRect();
+
+
+      if (itemRef.current) {
+        let finalItem = item.getBoundingClientRect();
+
+        // Invert: determine the delta between the first and last bounds of the element
+        // let deltaX = cachedItem.left - finalItem.left;
+        // let deltaW = cachedItem.width / finalItem.width;
+        // let deltaW = finalItem.width / cachedItem.width;
+        // let deltaH = cachedItem.height / finalItem.height;
+        let cachedX = itemRef?.current.offsetLeft;
+        let finalX = item.offsetLeft;
+        // let finalW = item.offsetWidth;
+        // let finalH = item.offsetHeight;
+
+        // styleObj.width = finalW;
+        // styleObj.height = finalH;
+        // setStyle(styleObj);
+        
+        // Play: animate the final element from its first bounds to its last bound
+        slide.animate(
+          [
+            {transform: `translateX(${finalX}px)` },
+          ],
+          {
+            duration: 150,
+            easing: 'ease-in',
```

Comment:
We may want to test the options, but I think `ease-out` is better for this use case.